yoke/service.py

```diff
@@ -268,9 +269,11 @@ def run_webserver(port, path):
     print('Starting webserver on ', port, path)
     class RH(HTTPRequestHandler):
         basepath = path
-    with socketserver.TCPServer(('', port), RH) as httpd:
-        httpd.serve_forever()
-
+    try:
+        with socketserver.TCPServer(('', port), RH) as httpd:
+            httpd.serve_forever()
+    except OSError:
+        exit()
```

Comment:
When does this fail? If it fails due to port being already taken, then I think that printing a message to the user would be better than silently quitting (unless httpd prints error on its own).

Reply:
Ah, sorry, I didn't notice it runs in a separate thread.

Actually, what do you think about moving `HTTPRequestHandler`, `check_webserver`, `run_webserver` etc to a seperate file?

Reply:
> When does this fail? If it fails due to port being already taken, then I think that printing a message to the user would be better than silently quitting (unless httpd prints error on its own).

It's really not that clear, I'll try to leave more comments in my next commits.

I think you already figured it out, but just in case: as it runs on a separate thread and I really didn't want to `import Queue` or something just for this, `service.py` checks just before opening the connection if the thread is still alive. Just once. If it's not, it assumes it fails due to port being already taken.

> Actually, what do you think about moving `HTTPRequestHandler`, `check_webserver`, `run_webserver` etc to a seperate file?

Gladly. It should be easy to do and it would make the code that much tidier.

I'll also move all the error handling (or at the very least, all the error messages) to `yoke/bin/yoke`.

Reply:
Okay, two new commits pushed. What do you think?